plugin/krypto.md

```diff
@@ -121,5 +121,42 @@ verifyKZGProof accepts a commitment, fields z and y, and a proof. It requires th
 ```k
     syntax Bool ::= verifyKZGProof ( Bytes, Bytes, Bytes, Bytes ) [function, hook(KRYPTO.verifyKZGProof)]
  // -----------------------------------------------------------------------------------------------------
+```
+
+BLS12-381 Hook Functions
+------------------------
+
+For all functions below, points with coordinates (0, 0) are considered the point at infinity.
+
+-   `BLS12G1Add` adds two points in the G1 group.
+-   `BLS12G2Add` adds two points in the G2 group.
+-   `BLS12G1Mul` multiplies a point with a scalar in the G1 group.
+-   `BLS12G2Mul` multiplies a point with a scalar in the G2 group.
+-   `BLS12PairingCheck` accepts a list of points in G1 and a list of points in G2 and
+    returns true if the pairing was succesful.
+-   `BLS12MapToG1` maps a field element to the G1 group.
+-   `BLS12MapToG2` maps a field element to the G2 group.
+
+If the coordinates for the points do not fit in 384 bits, or the scalars do not
+fit in 256 bits, or the field values do not fit in 384 bits, all BLS12_381
+functions evaluate to `#False`.
+
+```k
+    syntax G1Point ::= BLS12G1Add        ( G1Point, G1Point )  [function, hook(KRYPTO.bls12G1Add)]
+    syntax G2Point ::= BLS12G2Add        ( G2Point, G2Point )  [function, hook(KRYPTO.bls12G2Add)]
+    syntax G1Point ::= BLS12G1Mul        ( G1Point, Int )      [function, hook(KRYPTO.bls12G1Mul)]
+    syntax G2Point ::= BLS12G2Mul        ( G2Point, Int )      [function, hook(KRYPTO.bls12G2Mul)]
+    syntax Bool    ::= BLS12PairingCheck ( List, List)         [function, hook(KRYPTO.bls12PairingCheck)]
+    syntax G1Point ::= BLS12MapFpToG1    ( Int )               [function, hook(KRYPTO.bls12MapFpToG1)]
+    syntax G2Point ::= BLS12MapFp2ToG2   ( Int, Int )          [function, hook(KRYPTO.bls12MapFp2ToG2)]
+
+    syntax Bool    ::= BLS12G1InSubgroup ( G1Point )           [function, hook(KRYPTO.bls12G1InSubgroup)]
+    syntax Bool    ::= BLS12G2InSubgroup ( G2Point )           [function, hook(KRYPTO.bls12G2InSubgroup)]
+    syntax Bool    ::= BLS12G1OnCurve    ( G1Point )           [function, hook(KRYPTO.bls12G1OnCurve)]
+    syntax Bool    ::= BLS12G2OnCurve    ( G2Point )           [function, hook(KRYPTO.bls12G2OnCurve)]
+
+    syntax Int ::= "BLS12_FIELD_MODULUS"  [macro]
```

Comment:
You should consider making this an `alias` instead so at pretty-printing time it prints the easier-to-read version. This can be helpful for symbolic execution.

```suggestion
    syntax Int ::= "BLS12_FIELD_MODULUS"  [alias]
```

Reply:
Done.